tasks/init.yml

```diff
@@ -0,0 +1,31 @@
+---
+
+# The goal of the tasks in this file is to determine whether AtoM needs to be
+# initialized. The inconvenience is that the current mechanism to initialize
+# the database is the same used to flush an existing database.
+
+- name: "Look up database contents"
+  check_mode: "no"
+  changed_when: False
+  register: "db_lookup"
+  command: >
+    mysql \
+      --user={{ atom_config_db_username }} \
+      --password={{ atom_config_db_password }} \
+      --host={{ atom_config_db_hostname }} \
+      --port={{ atom_config_db_port }} \
+      --skip-column-names \
+      --execute="SELECT COUNT(*) FROM information_object WHERE id = 1;" \
+        {{ atom_config_db_name }}
```

Comment:
The tasks fails when database is empty:

```
TASK [artefactual-atom : Look up database contents] ***************************************************************************************************************************************************************
Friday 18 December 2020  12:09:56 +0100 (0:00:01.941)       0:11:12.175 ******* 
fatal: [atom1.centos.camps.accesstomemory.org]: FAILED! => {"changed": false, "cmd": ["mysql", "--user=atom-user", "--password=caTyPms.20kiJk$", "--host=127.0.0.1", "--port=3306", "--skip-column-names", "--execute=SELECT COUNT(*) FROM information_object WHERE id = 1;", "atom"], "delta": "0:00:00.039165", "end": "2020-12-18 11:09:57.446906", "msg": "non-zero return code", "rc": 1, "start": "2020-12-18 11:09:57.407741", "stderr": "mysql: [Warning] Using a password on the command line interface can be insecure.\nERROR 1146 (42S02) at line 1: Table 'atom.information_object' doesn't exist", "stderr_lines": ["mysql: [Warning] Using a password on the command line interface can be insecure.", "ERROR 1146 (42S02) at line 1: Table 'atom.information_object' doesn't exist"], "stdout": "", "stdout_lines": []}
``` 

A new check could be added, I tested:

```
---

# The goal of the tasks in this file is to determine whether AtoM needs to be
# initialized. The inconvenience is that the current mechanism to initialize
# the database is the same used to flush an existing database.

- name: "Look up database contents 1/2"
  check_mode: "no"
  changed_when: False
  register: "db_information_object_exists"
  command: >
    mysql \
      --user={{ atom_config_db_username }} \
      --password={{ atom_config_db_password }} \
      --host={{ atom_config_db_hostname }} \
      --port={{ atom_config_db_port }} \
      --skip-column-names \
      --execute="SHOW TABLES LIKE 'information_object';" \
        {{ atom_config_db_name }}
  ignore_errors: true
  failed_when: false

- name: "Look up database contents 2/2"
  check_mode: "no"
  changed_when: False
  register: "db_lookup"
  command: >
    mysql \
      --user={{ atom_config_db_username }} \
      --password={{ atom_config_db_password }} \
      --host={{ atom_config_db_hostname }} \
      --port={{ atom_config_db_port }} \
      --skip-column-names \
      --execute="SELECT COUNT(*) FROM information_object WHERE id = 1;" \
        {{ atom_config_db_name }}
  ignore_errors: true
  failed_when: false

- name: "Create uninitialized fact"
  set_fact:
    uninitialized: False

- name: "Determine if flush/init is needed 1/2"
  set_fact:
    uninitialized: True
  when:
    - "db_information_object_exists.rc == 0"
    - "db_information_object_exists.stdout_lines == []"

- name: "Determine if flush/init is needed 2/2"
  set_fact:
    uninitialized: True
  when:
    - "db_lookup.rc == 0"
    - "db_lookup.stdout_lines == ['0']"
```